v0/destinations/ga4/transform.js

```diff
@@ -34,6 +34,34 @@ const {
   GA4_PARAMETERS_EXCLUSION
 } = require("./utils");
 
+/**
+ * Returns GA4 client_id
+ * @param {*} message
+ * @param {*} clientIdFieldIdentifier
+ * @returns
+ */
+const getGA4ClientId = (message, { clientIdFieldIdentifier }) => {
+  let clientId;
+  // first we will search from webapp
+  if (clientIdFieldIdentifier) {
+    clientId = get(message, clientIdFieldIdentifier);
+  }
+  // if we don't find it from the config then we will fall back to the default search
+  if (!clientId) {
+    clientId =
+      getDestinationExternalID(message, "ga4ClientId") ||
+      get(message, "anonymousId") ||
+      get(message, "messageId");
+  }
+  return clientId;
```

Comment:
```suggestion
  return (
    get(message, clientIdFieldIdentifier) ||
    getDestinationExternalID(message, "ga4ClientId") ||
    get(message, "anonymousId") ||
    get(message, "messageId")
  );
```

Reply:
@akashrpo this change will fail test cases, and in order it to work i need to pass clientIdIdentifier to every test case